src/host/settings.hpp

```diff
@@ -25,6 +25,7 @@ constexpr unsigned short MIN_WINDOW_OPACITY = 0x4D; // 0x4D is approximately 30%
 
 #include "ConsoleArguments.hpp"
 #include "../inc/conattrs.hpp"
+#include "../renderer/inc/BlinkingState.hpp"

```

Comment:
I'm a bit uncomfortable that `settings.hpp` depends on something inside `renderer`. It should be the other way around, isn't' it. Or should `BlinkingState` be moved inside `conattrs.hpp`, just like `CursorType`?

Reply:
Yeah, now that you mention it, that doesn't seem right. But the `BlinkingState` class can't go in conattrs - that's part of conhost. WT can get away with using enums from there, but it couldn't access any actual code. `CursorType` and `ExtendedAttrs` should really be moved somewhere else.

Originally I had the conhost _blinkingState_ instance in the `RenderData` class (which was essentially the same for WT), and that's why it's part of the renderer lib. I only moved it to `Settings` because everything in conhost ends up going through the `Settings` class via `Globals::getConsoleInformation`, and getting the render data just became an unnecessary level of indirection.

But I'm thinking now that it could just as easily go in the `CONSOLE_INFORMATION` class (which derives from `Settings`). That would require moving the `LookupAttributeColors` method there too, but that's probably not a bad thing - it actually simplifies the code to a certain extent.